test/errtest.c

```diff
@@ -36,25 +36,40 @@ static int test_print_error_format(void)
     char *out = NULL, *p = NULL;
     int ret = 0, len;
     BIO *bio = NULL;
+    int syserr = EPERM;
+    int reasoncode;
 
-    BIO_snprintf(expected, sizeof(expected), expected_format, OPENSSL_FUNC);
+    ERR_PUT_error(ERR_LIB_SYS, 0, syserr, "errtest.c", 30);
+    reasoncode = ERR_GET_REASON(ERR_peek_error());
 
-    if (!TEST_ptr(bio = BIO_new(BIO_s_mem())))
-        return 0;
+    if (reasoncode != syserr) {
+        TEST_skip("libcrypto didn't register EPERM (%d) correctly (%d).  "
+                  "This is a known limitation",
+                  syserr, reasoncode);
+        ERR_clear_error();
+    } else {
+        char buf[256];
+
+        OPENSSL_strlcpy(buf, strerror(syserr), sizeof(buf));
+        BIO_snprintf(expected, sizeof(expected), expected_format,
+                     OPENSSL_FUNC, buf);
```

Comment:
You don't really need the `buf`, simply taking `strerror(syserr)` as arg of `BIO_snprintf()`.

Reply:
Good point

Reply:
Done